### ISSUE TITLE
Device mismatch in ModularEvasionAttackMinDistance

### ISSUE THREAD

**Issue by maurapintor** [pralab/secml-torch#198]
The attacks need to be tested with a different device than `cpu`. There are several places in the attacks and in the loss functions that are missing a `.to(device)`.

Example to reproduce: 
```python
model = model.to("cuda")
model.eval()
attack = FMN(
            perturbation_model=LpPerturbationModels.LINF,
            num_steps=10,
            step_size=1,
            y_target=None,
            backend=Backends.NATIVE,
        )
```

Note: Possibly also test trackers.

**Comment by maurapintor** [comment]
Might be related to #173, #117.
